File: BKabanApi/BKabanApi/Controllers/TaskController.cs
using BKabanApi.Models;
using BKabanApi.Models.DB;
using BKabanApi.Utils;
using Microsoft.AspNetCore.Mvc;

namespace BKabanApi.Controllers
{
    [Route("api/task")]
    [ApiController]
    public class TaskController : ControllerBase
    {
        private readonly ITaskRepository _taskRepository;

        public TaskController(ITaskRepository taskRepository)
        {
            _taskRepository = taskRepository;
        }

        [HttpPost]
        public ActionResult CreateTask(TaskModelColumnLink task)
        {
            int? userId;

            if ((userId = AuthHelper.GetUserId(HttpContext)) == null)
            {
                return Unauthorized();
            }

            int? taskId = _taskRepository.CreateTask((int) userId, task);

            if (taskId == null)
            {
                return StatusCode(403);
            }

            return Ok(new { id = taskId });
        }

        [HttpPut]
        public ActionResult UpdateTask(TaskModel task)
        {
            int? userId;

            if ((userId = AuthHelper.GetUserId(HttpContext)) == null)
            {
                return Unauthorized();
            }

            if (task.Id == null || task.Name == null && task.Description == null)
            {
                return BadRequest();
            }

            bool result = _taskRepository.UpdateTask((int) userId, task);

            if (result)
            {
                return Ok();
            }

            return StatusCode(403);
        }

        [HttpPut("reorder")]
        public ActionResult MoveTask(TaskModelWithPositionAndNewColumn task)
        {
            int? userId;

            if ((userId = AuthHelper.GetUserId(HttpContext)) == null)
            {
                return Unauthorized();
            }

            if (task.Id == null)
            {
                return BadRequest();
            }

            bool result = _taskRepository.UpdateTaskPositionAndColumn((int)userId, task);

            if (result)
            {
                return Ok();
            }

            return StatusCode(403);
        }

        [HttpDelete("{id}")]
        public ActionResult DeleteTask(int id)
        {
            int? userId;

            if ((userId = AuthHelper.GetUserId(HttpContext)) == null)
            {
                return Unauthorized();
            }

            bool result = _taskRepository.DeleteTask((int)userId, id);

            if (result)
            {
                return Ok();
            }

            return StatusCode(403);
        }
    }
}

File: BKabanApiJava/src/main/java/edu/bkaban/controllers/TaskController.java
package edu.bkaban.controllers;

import edu.bkaban.models.IdDTO;
import edu.bkaban.models.task.TaskModel;
import edu.bkaban.models.task.TaskModelColumnLink;
import edu.bkaban.models.task.TaskModelWithPositionAndNewColumn;
import edu.bkaban.repositories.TaskRepository;
import edu.bkaban.services.AuthService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.HttpStatus;
import org.springframework.http.MediaType;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

import javax.servlet.http.HttpSession;
import javax.validation.Valid;
import java.sql.SQLException;

@RestController
@RequestMapping("/api/task")
public class TaskController {

    @Autowired
    private TaskRepository _taskRepository;

    @Autowired
    private AuthService _authService;

    @PostMapping
    private ResponseEntity createTask(@Valid @RequestBody TaskModelColumnLink task, HttpSession session) throws SQLException {
        Integer userId;
        if ((userId = _authService.getUserId(session)) == null) {
            return new ResponseEntity(HttpStatus.UNAUTHORIZED);
        }

        Integer taskId = _taskRepository.createTask(userId, task);
        if (taskId == null) {
            return new ResponseEntity(HttpStatus.FORBIDDEN);
        }

        return ResponseEntity.ok(new IdDTO(taskId));
    }

    @PutMapping
    private ResponseEntity updateTask(@Valid @RequestBody TaskModel task, HttpSession session) throws SQLException {
        Integer userId;
        if ((userId = _authService.getUserId(session)) == null) {
            return new ResponseEntity(HttpStatus.UNAUTHORIZED);
        }

        if (task.getId() == null || task.getName() == null && task.getDescription() == null) {
            return new ResponseEntity(HttpStatus.BAD_REQUEST);
        }

        boolean isOk = _taskRepository.updateTask(userId, task);
        if (isOk) {
            return new ResponseEntity(HttpStatus.OK);
        }

        return new ResponseEntity(HttpStatus.FORBIDDEN);
    }

    @PutMapping("/reorder")
    private ResponseEntity moveTask(@Valid @RequestBody TaskModelWithPositionAndNewColumn task, HttpSession session) throws  SQLException {
        Integer userId;
        if ((userId = _authService.getUserId(session)) == null) {
            return new ResponseEntity(HttpStatus.UNAUTHORIZED);
        }

        if (task.getId() == null) {
            return new ResponseEntity(HttpStatus.BAD_REQUEST);
        }

        boolean isOk = _taskRepository.updateTaskPositionAndColumn(userId, task);
        if (isOk) {
            return new ResponseEntity(HttpStatus.OK);
        }

        return new ResponseEntity(HttpStatus.FORBIDDEN);
    }

    @DeleteMapping("/{id}")
    private ResponseEntity deleteTask(@PathVariable int id, HttpSession session) throws SQLException {
        Integer userId;
        if ((userId = _authService.getUserId(session)) == null) {
            return new ResponseEntity(HttpStatus.UNAUTHORIZED);
        }

        boolean isOk = _taskRepository.deleteTask(userId, id);
        if (isOk) {
            return new ResponseEntity(HttpStatus.OK);
        }

        return new ResponseEntity(HttpStatus.FORBIDDEN);
    }
}
